tests/functional/conftest.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 Martin Kalcok
+# See LICENSE file for licensing details.
+"""Pytest fixtures for functional tests."""
+#  pylint: disable=W0621
+
+import logging
+import tempfile
+from pathlib import Path
+
+import pytest
+from kubernetes import client, config
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="session")
+def namespace() -> str:
+    """Return namespace used for functional tests."""
+    return "default"
+
+
+@pytest.fixture(scope="module")
+async def kube_config(ops_test: OpsTest) -> Path:
+    """Return path to the kube config of the tested Kubernetes cluster.
+
+    Config file is fetched from kubernetes-master unit and stored in the temporary file.
+    """
+    k8s_master = ops_test.model.applications["kubernetes-master"].units[0]
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        kube_config_file = Path(tmp_dir).joinpath("kube_config")
+
+        cmd = "juju scp {}:config {}".format(k8s_master.name, kube_config_file).split()
+        return_code, _, std_err = await ops_test.run(*cmd)
+        assert return_code == 0, std_err
+        yield kube_config_file
+
+
+@pytest.fixture()
+async def cleanup_k8s(kube_config, namespace: str):
+    """Cleanup kubernetes resources created during test."""
+    yield  # act only on teardown
+    config.load_kube_config(str(kube_config))
+
+    pod_prefixes = ["read-test-ceph-", "write-test-ceph"]
+    pvc_prefix = "pvc-test-"
+    core_api = client.CoreV1Api()
+
+    for pod in core_api.list_namespaced_pod(namespace).items:
+        pod_name = pod.metadata.name
+        if any(pod_name.startswith(prefix) for prefix in pod_prefixes):
```

Comment:
If "not any" is used, calling `continue`, you could get rid of the indentation on the rest of the code.

Reply:
Imo it reads bit more naturally this way and the indentation is not too deep. I'd leave the original implementation.